crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -458,13 +539,39 @@ async fn router(state: StateType) -> Router {
 /// accepts an address to listen on (which could be a wildcard like 0.0.0.0:0), and a set of
 /// initial accounts to initialize the state.
 /// returns the address that the server is listening on (with any input wildcards resolved).
-pub async fn run(
+pub async fn serve(
     address: SocketAddr,
     config: RpcHardhatNetworkConfig,
-    genesis_accounts: HashMap<Address, AccountInfo>,
-) -> Result<SocketAddr, std::io::Error> {
+    genesis_accounts: Option<HashMap<Address, AccountInfo>>,
+) -> Result<
+    axum::Server<hyper::server::conn::AddrIncoming, axum::routing::IntoMakeService<Router>>,
```

Comment:
I think not leaking the `axum::Server` type would also avoid needing the hyper dependency

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4057/commits/77e9cc09bf764b2d9448fc35548afe72b6f61cff
though, it didn't eliminate the dependency, as we're still keeping a private instance.